sdv/single_table/base.py

```diff
@@ -343,12 +352,26 @@ def get_parameters(self):
 
         return instantiated_parameters
 
-    def get_metadata(self):
-        """Return the ``Metadata`` for this synthesizer."""
-        if isinstance(self.metadata, SingleTableMetadata):
-            table_name = getattr(self, '_table_name', None)
-            return Metadata.load_from_dict(self.metadata.to_dict(), table_name)
-        return self.metadata
+    def get_metadata(self, version='original'):
+        """Get the metadata, either original or modified after applying CAG patterns.
+
+        Args:
+            version (str, optional):
+                The version of metadata to return, must be one of 'original' or 'modified'. If
+                'original', will return the original metadata used to instantiate the
+                synthesizer. If 'modified', will return the modified metadata after applying this
+                synthesizer's CAG patterns. Defaults to 'original'.
+        """
+        if version not in ('original', 'modified'):
+            raise ValueError(
+                f"Unrecognized version '{version}', please use 'original' or 'modified'."
+            )
+
+        table_name = getattr(self, '_table_name', None)
+        if hasattr(self, '_original_metadata') and version == 'original':
+            return Metadata.load_from_dict(self._original_metadata.to_dict(), table_name)
```

Comment:
If `_original_metadata` is already a `Metadata` object this is unnecessary.

Reply:
`get_metadata` has to return a copy.

Reply:
That's a good catch. In that case though, returning a deepcopy of the object makes the intention clearer. Either way, this change should probably be replicated on the other two `get_metadata` methods [here](https://github.com/sdv-dev/SDV/blob/feature/single-table-CAG/sdv/multi_table/base.py#L203-L209) and [here](https://github.com/datacebo/SDV-Enterprise/blob/eb3ac78b1a6a4b685756251a85abb9a371be64fe/sdv_enterprise/sdv/multi_table/cag/base.py#L55-L61).

Reply:
Good catch, I will add it to those files asap.

Reply:
Done in [8cceac7](https://github.com/sdv-dev/SDV/pull/2503/commits/8cceac7a3149dc487035a4f51c8bba3b1488a3a2)